lms/djangoapps/courseware/access_utils.py

```diff
@@ -103,3 +112,57 @@ def check_course_open_for_learner(user, course):
     if COURSE_PRE_START_ACCESS_FLAG.is_enabled():
         return ACCESS_GRANTED
     return check_start_date(user, course.days_early_for_beta, course.start, course.id)
+
+
+def check_enrollment(user, course):
+    """
+    Check if the course requires a learner to be enrolled for access.
+
+    Returns:
+        AccessResponse: Either ACCESS_GRANTED or EnrollmentRequiredAccessError.
+    """
+    if check_public_access(course, [COURSE_VISIBILITY_PUBLIC]):
+        return ACCESS_GRANTED
+
+    if CourseEnrollment.is_enrolled(user, course.id):
+        return ACCESS_GRANTED
+
+    return EnrollmentRequiredAccessError()
+
+
+def check_authentication(user, course):
+    """
+    Grants access if the user is authenticated, or if the course allows public access.
+
+    Returns:
+        AccessResponse: Either ACCESS_GRANTED or AuthenticationRequiredAccessError
+    """
+    if user.is_authenticated:
+        return ACCESS_GRANTED
+
+    if check_public_access(course, [COURSE_VISIBILITY_PUBLIC]):
+        return ACCESS_GRANTED
+
+    return AuthenticationRequiredAccessError()
+
+
+def check_public_access(course, visibilities):
```

Comment:
This function was moved from `allow_public_access` in courseware/courses.py below.  It's identical _except_ that it now returns an AccessResponse object.  These objects can be evaluated as booleans thanks to their `__bool__` methods.  The return value of this function is always evaluated as a boolean anyway, so this should be a seamless change.